docs: add repo readme and install instructions (#69)
UpCloudLtd/upcloud-whmcs-module: docs/INSTALL.md

7. **Custom Plans**: Once set by admin during product setup, users can customize their plans and select them for deployment after making a payment.

8. **Cron Job**: The administrator must schedule cron jobs to run every 5 minutes to generate bandwidth usage graphs.

9. **Pricing Adjustment**: While UpCloud has predefined prices for Windows, CPU, RAM, and storage, you have the flexibility to adjust them according to your specific requirements.

### Client Features:

1. **VM Information**: Clients can view basic VM information such as VPS, VNC, IP addresses details.

2. **VNC Management**: Clients can enable/disable VNC access.

3. **Password Management**: Clients can change their VNC passwords.

4. **Reverse PTR**: Clients can view and set reverse PTR records.

5. **VM Control**: Clients can start, stop, and restart their VMs.

6. **Configuration Modification**: Clients can modify server configurations to suit their needs.

7. **Bandwidth Usage Graph**: Clients can view bandwidth usage graphs for intervals spanning 24 hours, a week, a month, and a year.

## Installation

### Downloading and Uploading to WHMCS

1. Obtain the module from our GitHub repository.

> 2. Unpack the module on your local system and transfer the "upCloudVps" folder to your WHMCS directory named "/path-to-your-WHMCS/modules/servers/" on your WHMCS installation.

Reviewer: We may want to adjust this some when we have some actual release zips or the like available. (The zip will likely contain the needed dir structure including parent dirs.)

### Obtaining API Credentials

> 1. Access the URL "https://hub.upcloud.com/dashboard" and log in using your credentials.

Reviewer: ```suggestion
1. Access the UpCloud Hub at  https://hub.upcloud.com and log in using your credentials.
```

> 2. Navigate to the "People" section from the left sidebar or directly visit "https://hub.upcloud.com/people/accounts".
> 3. Proceed to create a sub-account by selecting the "Create subaccount" button or visit "https://hub.upcloud.com/people/accounts/new".

Reviewer: Not sure if the direct links add value here

```suggestion
2. Navigate to the "People" section from the left sidebar.
3. Proceed to create a sub-account by selecting the "Create subaccount" button.
```

3. Enter the Hostname or IP Address as: api.upcloud.com (you can assign any name).
4. For Username, input your API username created in the previous step.
5. For Password, input your API password created in the previous step.
6. Proceed by clicking on "Test Connection" or "Continue Anyway".

### Creating a Server Group

1. Click Create New Group under the Options heading.
2. Enter a name for your group.
3. Select the servers you want to assign (previously created) to this group in the box on the left.
4. Click Add to move them to the box on the right, which contains the servers for this group.
5. Click Save Changes to complete the process.

## Configuration

### Creating a Product Group

1. Navigate to Configuration > System Settings > Product/Services

![Producs & Services](images/products_services.png)

2. Click on Create Product Group

![Product Group](images/product_group.png)

3. Provide a Product Group name and adjust all other settings based on your requirements

![Product Config](images/product_config.png)

4. Click on Save Changes

### Creating a New Product

1. Click on Create a New Product

![New Product](images/new_product.png)
2. Configure the following details:

>    - **Product Type**: Select as "Other" (recommended to avoid prompting customers for Hostname, Password, NS1 & NS2 prefix during order process)

Reviewer: This has some other positive (~required) consequences, not sure if enumerating them here is necessary

```suggestion
   - **Product Type**: Select as "Other"
```

   - **Product Group**: Select the product group that you created
   - **Product Name**: Provide your product name
   - **Module**: UpCloud VPS
   - **Create as Hidden**: Off
3. Click on Save Changes

### Module Settings

On the next page:
1. Verify the Product Type
2. Set Welcome Email to "Dedicated/VPS Server Welcome Email"
3. Go to the Module Settings Tab
4. Select the required details:
   - Module Name
   - Server group
   - Default Location
   - Plan
   - Template
5. If for Plans you selected "Custom", then users will have the ability to choose their own CPU, RAM, and Storage space based on which VMs will be created.

![Server Plan Config](images/server_plan_config.png)

6. Decide whether the product requires manual approval from admins or is automatically deployed upon order.
7. Click the Save Changes button.

![Server Plan Ready](images/server_plan_ready.png)

This will create custom fields named (`instanceId`, `ssh_key` & `userData`) as shown below:

![Custom Fields](images/custom_fields.png)

> ### Configurable Options

Reviewer: Configurable option sets were recently changed in #68 , an update to this whole section seems to be in order.

Author: Updated to match the current implementation

- **Memory**: Users can choose the memory of a VPS, ranging from 4GB to 64GB.
- **vCPU**: Users can choose the vCPU of a VPS, ranging from 2 cores to 24 cores.
- **Storage**: Users can choose the storage of a VPS, ranging from 50GB to 460GB.

![Configurable Options Custom](images/configurable_options_custom.png)

## Adding Existing VPS to Users

To add an existing UpCloud VPS to a user's account:
1. Edit the user's product
2. Within the custom field named "Instance Id", add your VPS UUID that you can obtain from UpCloud Hub

![Existing VPS](images/existing_vps.png)

## Additional Configuration

### Product/Service -> Custom Fields

- **User Data/Cloud-init**: If provided by the user, these scripts will initiate after VM creation to kickstart the initialization process.
- **SSH Key**: If supplied by the user during VM creation, it will be utilized for user authentication instead of a password; otherwise, the system will default to using a password for authentication. Note that some templates only support SSH keys.
- **Instance Id**: This field is used to assign a user with a VPS

> ### System Settings -> Configurable Options

Reviewer: An update to this wrt recent changes re #68 would be in order here, too.

Author: Thanks, I moved this to the section above about product config & configurable options

### Configurable Options

The module will automatically create Configurable Options when you create your first product. You can select option groups that apply to your product.

To adjust these options:

1. Go to System Settings -> Configurable Options
2. Edit the newly created Options

![Configurable Options](images/configurable_options_normal.png)

Through these configurations you can:

- Set custom pricing for backup options
- Configure available locations
- Configure available custom plans
- Adjust prices for available images (such as Windows)
- Backup and Location: set custom pricing for backup options, and configure available locations

> If custom plans are utilised:

Reviewer: Could note here that custom plans are an account feature that requires special provisioning, contact support etc.

Author: Added!
